src/core/Header/HeaderButton.tsx

```diff
@@ -78,11 +85,21 @@ export const HeaderButton = (props: HeaderButtonProps) => {
     ...rest,
   };
 
-  return menuItems ? (
-    <DropdownButton {...buttonProps} menuItems={menuItems} />
-  ) : (
-    <Button {...buttonProps} />
-  );
+  if (menuItems && props.onClick) {
+    return (
+      <SplitButton
+        {...buttonProps}
+        menuItems={menuItems}
+        onClick={() => buttonProps.onClick}
+      >
+        {buttonProps.children}
+      </SplitButton>
+    );
+  }
+  if (menuItems) {
+    return <DropdownButton {...buttonProps} menuItems={menuItems} />;
+  }
+  return <Button {...buttonProps} />;
```

Comment:
This doesn't work?
```suggestion
  return menuItems ? (
    isSplitButton ? (
      <SplitButton {...buttonProps} menuItems={menuItems} 
    ) : (
      <DropdownButton {...buttonProps} menuItems={menuItems} />
    )
  ) : (
    <Button {...buttonProps} />
  );
```

The main things here is that `buttonProps` already includes onClick and children, so you don't need to specify them.

Reply:
I prefer the current one. I really hate ternary inside ternary 😃 Even it is considered best practice to return component earlier if some condition is met.

Reply:
Split button props has mandatory onClick and children 
![image](https://user-images.githubusercontent.com/83585998/129029209-ab59741e-72b0-47ef-a8b3-febd8e21ee85.png)
I get an error when not providing those 
![image](https://user-images.githubusercontent.com/83585998/129029311-2fc4be1e-012d-47e8-b829-9e9d22e9a2bc.png)

Reply:
Maybe you can change the check to `if (buttonProps.menuItems && buttonProps.onClick)`.